File: modules/matterhorn-textextractor-tesseract/src/main/java/org/opencastproject/textextractor/tesseract/TesseractLine.java
/**
 *  Copyright 2009, 2010 The Regents of the University of California
 *  Licensed under the Educational Community License, Version 2.0
 *  (the "License"); you may not use this file except in compliance
 *  with the License. You may obtain a copy of the License at
 *
 *  http://www.osedu.org/licenses/ECL-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an "AS IS"
 *  BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 *  or implied. See the License for the specific language governing
 *  permissions and limitations under the License.
 *
 */
package org.opencastproject.textextractor.tesseract;

import org.opencastproject.textextractor.api.TextLine;

import java.awt.Rectangle;

/**
 * Representation of a line of text extracted from an image.
 */
public class TesseractLine implements TextLine {

  /** The text */
  protected String text = null;

  /**
   * Creates a representation for a piece of text
   *
   * @param word
   *          the extracted text
   */
  public TesseractLine(String line) {
    this.text = line;
  }

  /**
   * {@inheritDoc}
   *
   * @see org.opencastproject.textextractor.api.TextLine#getText()
   */
  @Override
  public String getText() {
    return text;
  }

  /**
   * {@inheritDoc}
   *
   * @see org.opencastproject.textextractor.api.TextLine#getBoundaries()
   */
  @Override
  public Rectangle getBoundaries() {
    return null;
  }

  /**
   * {@inheritDoc}
   *
   * @see java.lang.Object#toString()
   */
  @Override
  public String toString() {
    return text;
  }

}

File: modules/textextractor-tesseract/src/main/java/org/opencastproject/textextractor/tesseract/TesseractLine.kt
/**
 * Licensed to The Apereo Foundation under one or more contributor license
 * agreements. See the NOTICE file distributed with this work for additional
 * information regarding copyright ownership.
 *
 *
 * The Apereo Foundation licenses this file to you under the Educational
 * Community License, Version 2.0 (the "License"); you may not use this file
 * except in compliance with the License. You may obtain a copy of the License
 * at:
 *
 * http://opensource.org/licenses/ecl2.txt
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations under
 * the License.
 *
 */

package org.opencastproject.textextractor.tesseract

import org.opencastproject.textextractor.api.TextLine

import java.awt.Rectangle

/**
 * Representation of a line of text extracted from an image.
 */
class TesseractLine
/**
 * Creates a representation for a piece of text
 *
 * @param line
 * the extracted text
 */
(line: String) : TextLine {

    /** The text  */
    /**
     * {@inheritDoc}
     *
     * @see org.opencastproject.textextractor.api.TextLine.getText
     */
    override var text: String? = null
        protected set(value: String?) {
            super.text = value
        }

    /**
     * {@inheritDoc}
     *
     * @see org.opencastproject.textextractor.api.TextLine.getBoundaries
     */
    override val boundaries: Rectangle
        get() = null

    init {
        this.text = line
    }

    /**
     * {@inheritDoc}
     *
     * @see java.lang.Object.toString
     */
    override fun toString(): String? {
        return text
    }

}
